code/game/objects/items/devices/scanners.dm

```diff
@@ -37,7 +54,15 @@ GENE SCANNER
 		STOP_PROCESSING(SSobj, src)
 
 /obj/item/t_scanner/attack_self(mob/user)
-	toggle_on()
+	if(istype(src, /obj/item/t_scanner/polytool))
```

Comment:
Można to przenieść do czegoś w stylu
```
	/obj/item/t_scanner/polytool/attack_self(mob/user)
		playsound(get_turf(user), 'sound/items/change_jaws.ogg', 50, 1)
		var/obj/item/wirecutters/power/polytool/polycutter = new /obj/item/wirecutters/power/polytool(drop_location())
		to_chat(user, "<span class='notice'>You change polytool mode to wirecutter.</span>")
		qdel(src)
		user.put_in_active_hand(polycutter)
		return
```

Reply:
Można, tylko nie wiem jaką to daje przewagę nad tym rozwiązaniem które jest

Reply:
Minimalne przyspieszenie + czytelność kodu na przyszłość.
W obecnej postaci kod sprawdza czy nasze narzędzie to Polytool, nawet gdy używamy zwykłego T-raya.